gson/pom.xml

```diff
@@ -222,4 +238,17 @@
       </plugin>
     </plugins>
   </build>
+  <profiles>
+    <profile>
+      <id>JDK17</id>
+      <activation>
+        <jdk>[17,)</jdk>
+      </activation>
+      <properties>
+        <javaVersion>17</javaVersion>
+        <excludeTestCompilation></excludeTestCompilation>
+        <extraLintSuppressions>,-exports,-missing-explicit-ctor,-removal</extraLintSuppressions>
```

Comment:
Would be good to fix these issues, I think the warnings are reasonable:
- `TypeAdapter` should have an explicit no-args constructor
- `TypeAdapters` should suppress warnings for usage of `AccessController.doPrivileged` (related to `SecurityManager` deprecation)
- `Gson.excluder()` should have `@SuppressWarnings("exports")` (see also #1903)

For the module related warnings for test compilation, it looks like Maven erroneously considers `module-info.java` also for the test classes? Might be good to investigate that. Or then at least only disable these warnings for test compilation.

Reply:
I agree. I was going to do that in a separate PR. The problems don't seem to show up all the time, and I'm not sure I fully understand why these started cropping up with this PR. Maybe the version of JDK 17 in the CI has changed very recently?